src/nvim/api/win_config.c

```diff
@@ -479,16 +479,18 @@ void nvim_win_set_config(Window window, Dict(win_config) *config, Error *err)
     }
 
     bool to_split_ok = false;
+    int dir = 0;
     // If we are moving curwin to another tabpage, switch windows *before* we remove it from the
     // window list or remove its frame (if non-floating), so it's valid for autocommands.
     const bool curwin_moving_tp = win == curwin && parent && win_tp != parent_tp;
     if (curwin_moving_tp) {
-      if (was_split) {
-        int dir;
-        win_goto(winframe_find_altwin(win, &dir, NULL, NULL));
-      } else {
-        win_goto(win_float_find_altwin(win, NULL));
+      win_T *target_wp = was_split ? winframe_find_altwin(win, &dir, NULL,
+                                                          NULL) : win_float_find_altwin(win, NULL);
+      if (!target_wp) {
+        return;
```

Comment:
Rather than a silent failure, make this error with something like `Failed to switch away from window %d` below.